src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class ModLoader : Control
+{
+    [Export]
+    public NodePath UnloadedItemListPath;
+
+    [Export]
+    public NodePath ModInfoNamePath;
+
+    [Export]
+    public NodePath ModInfoAuthorPath;
+
+    [Export]
+    public NodePath ModInfoVersionPath;
+
+    [Export]
+    public NodePath ModInfoDescriptionPath;
+
+    [Export]
+    public NodePath LoadedItemListPath;
+
+    [Export]
+    public NodePath ConfirmationPopupPath;
+
+    private ItemList unloadedItemList;
+    private ItemList loadedItemList;
+
+    // Labels For The Mod Info Box
+    private Label modInfoName;
+    private Label modInfoAuthor;
+    private Label modInfoVersion;
+    private Label modInfoDescription;
+
+    private ConfirmationDialog confirmationPopup;
+
+    private List<ModInfo> modList = new List<ModInfo>();
+    private List<ModInfo> loadedModList = new List<ModInfo>();
+
+    [Signal]
+    public delegate void OnModLoaderClosed();
+
+    public override void _Ready()
+    {
+        unloadedItemList = GetNode<ItemList>(UnloadedItemListPath);
+        loadedItemList = GetNode<ItemList>(LoadedItemListPath);
+
+        modInfoName = GetNode<Label>(ModInfoNamePath);
+        modInfoAuthor = GetNode<Label>(ModInfoAuthorPath);
+        modInfoVersion = GetNode<Label>(ModInfoVersionPath);
+        modInfoDescription = GetNode<Label>(ModInfoDescriptionPath);
+
+        confirmationPopup = GetNode<ConfirmationDialog>(ConfirmationPopupPath);
+
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
```

Comment:
Duplicated code. Should probably be made into a method for easily refreshing the list of available mods.